### ISSUE TITLE
range-diff: allow comparing branches to mboxes

### ISSUE THREAD

**Issue by dscho** [gitgitgadget/git#207]
A neat trick that could come in real handy at times would be to teach `git range-diff` to compare a local branch to a patch series that is provided as an mbox, e.g. to use `git range-diff` to see
whether a given patch series on the mailing list has made it into git.git's `pu` branch yet.

We could introduce `range-diff mbox:<file> A..B` for that, for example.

This would still require some scripting to e.g. pull down the mbox files of a patch series from https://public-inbox.org/git.

But it would solve the problem where current `range-diff` *requires* local branches, and therefore the user would have to find a commit where the patch series actually applies first.

**Comment by LemmingAvalanche** [comment]
What looks like the last patch series: https://lore.kernel.org/git/pull.1420.git.1668536405563.gitgitgadget@gmail.com/#t

**Comment by dscho** [comment]
> What looks like the last patch series: https://lore.kernel.org/git/pull.1420.git.1668536405563.gitgitgadget@gmail.com/#t

@LemmingAvalanche yes, I have updated this branch a couple of times in the meantime but never got around to polish and submit a new iteration.

**Comment by dscho** [comment]
> > What looks like the last patch series: https://lore.kernel.org/git/pull.1420.git.1668536405563.gitgitgadget@gmail.com/#t
> 
> [@LemmingAvalanche](https://github.com/LemmingAvalanche) yes, I have updated this branch a couple of times in the meantime but never got around to polish and submit a new iteration.

For the record: As of time of writing, 15d1b8a72305e63e14a8c29c5d830f6033494ab6 is the tip of the latest revision I have (and use, locally).
